packages/components/progress-bar/src/progress-bar.tsx

```diff
@@ -0,0 +1,295 @@
+      >
+        {props.height === '10' ? (
+          <Text.Detail tone={props.isInverted ? 'inverted' : undefined}>
+            {getLabel(props.label)}
+          </Text.Detail>
+        ) : (
+          <Text.Body
+            as="span"
+            tone={props.isInverted ? 'inverted' : undefined}
+            fontWeight="medium"
+          >
+            {getLabel(props.label)}
+          </Text.Body>
+        )}
+      </div>
+    </Constraints.Horizontal>
+  );
+};
+
+const progressPulse = keyframes`
+  0% {
+    background-position: 100% 0;
+  }
+  100% {
+    background-position: -100% 0;
+  }
+`;
+
+const Bar = (
+  props: Pick<
+    TProgressBarProps,
+    'progress' | 'height' | 'barWidth' | 'isInverted' | 'isAnimated'
+  >
+) => {
+  return (
+    <Constraints.Horizontal max={props.barWidth}>
+      <div
+        css={css`
+          background-color: ${props.isInverted
+            ? 'rgba(255, 255, 255, 0.4)'
+            : designTokens.colorNeutral90};
+          border-radius: ${designTokens.spacingL};
+          height: ${heightPerScale[props.height ?? '20']};
+          overflow: hidden;
+        `}
+        role="progressbar"
+        aria-valuenow={props.progress}
+        aria-label="Progress bar"
+        {...filterAriaAttributes(props)}
+      >
+        <div
+          css={css`
+            width: ${props.progress}%;
+            transition: width 500ms ease-in-out;
+            display: block;
+            height: ${heightPerScale[props.height ?? '20']};
+            background: ${props.isInverted
+              ? designTokens.colorSurface
+              : `linear-gradient(
+                to right,
+                #00E5CB,
+                ${designTokens.colorPrimary25},
+                #00E5CB
+              )`};
+            background-size: 200% 100%;
+            animation: ${props.isAnimated && !props.isInverted
+              ? css`
+                  ${progressPulse} 2s linear infinite
+                `
+              : 'none'};
+            border-radius: ${designTokens.spacingL};
+          `}
+        />
+      </div>
+    </Constraints.Horizontal>
+  );
+};
+
+const ProgressBar = (props: TProgressBarProps) => {
+  switch (props.labelPosition) {
+    case 'bottom':
+      return (
+        <SpacingsStack scale="m" alignItems="center">
+          <Bar
+            progress={props.progress}
+            barWidth={props.barWidth}
+            isInverted={props.isInverted}
+            height={props.height}
+            isAnimated={props.isAnimated}
+          />
+          <ProgressBarLabel
+            label={props.label}
+            labelWidth={props.labelWidth}
+            labelPosition={props.labelPosition}
+            isInverted={props.isInverted}
+            height={props.height}
+          />
+        </SpacingsStack>
+      );
+    case 'left':
+      return (
+        <SpacingsInline scale="m" alignItems="center">
+          <div
+            css={css`
+              text-align: right;
```

Comment:
I don't get when we need this style.
Could you please share the use case it is here for? 🙏